pkg/kubectl/cmd/create_pvc.go

```diff
@@ -0,0 +1,222 @@
+func NewCmdCreatePersistentVolumeClaim(f cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	c := &CreatePVCOptions{
+		Out: cmdOut,
+	}
+	cmd := &cobra.Command{
+		Use:     "persistentvolumeclaim NAME --access-mode=accessmode --capacity=capacity [--storage-class=storageclass] [--volume-name=volumename] [--dry-run]",
+		Short:   pvcLong,
+		Long:    pvcLong,
+		Example: pvcExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(c.Complete(f, cmd, args))
+			cmdutil.CheckErr(c.Validate())
+			cmdutil.CheckErr(c.RunCreatePVC())
+		},
+	}
+	cmdutil.AddApplyAnnotationFlags(cmd)
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddPrinterFlags(cmd)
+	cmdutil.AddDryRunFlag(cmd)
+
+	cmd.Flags().StringSlice("access-mode", []string{}, "AccessMode contains the desired access modes the volume should have.")
+	cmd.Flags().String("capacity", "", "Capacity represents the minimum storage capacity required.")
+	cmd.Flags().String("storage-class", "", "Name of the StorageClass required by the claim.")
+	cmd.Flags().StringVar(&c.VolumeName, "volume-name", "", "VolumeName is the binding reference to the PersistentVolume backing this claim.")
+
+	return cmd
+}
+
+func (c *CreatePVCOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	name, err := NameFromCommandArgs(cmd, args)
+	if err != nil {
+		return err
+	}
+	c.Name = name
+
+	// Transform access mode abbreviation to complete access mode and remove duplicate access mode.
+	accessModes := []api.PersistentVolumeAccessMode{}
+	modes := cmdutil.GetFlagStringSlice(cmd, "access-mode")
+	for _, m := range modes {
+		var accessMode api.PersistentVolumeAccessMode
+		switch m {
+		case "RWX", string(api.ReadWriteMany):
+			accessMode = api.ReadWriteMany
+		case "RWO", string(api.ReadWriteOnce):
+			accessMode = api.ReadWriteOnce
+		case "ROX", string(api.ReadOnlyMany):
+			accessMode = api.ReadOnlyMany
+		default:
+			return fmt.Errorf("invalid access mode: %s", m)
+		}
+
+		containsMode := false
+		for _, a := range accessModes {
+			if a == accessMode {
+				containsMode = true
+				break
+			}
+		}
+
+		if !containsMode {
+			accessModes = append(accessModes, accessMode)
+		}
+	}
+	c.AccessModes = accessModes
+
+	// If user does not provide storage-class option, then StorageClass will be set as nil.
+	// Otherwise, StorageClass will be the value set by user, including empty string.
+	storageClassFlag := cmd.Flags().Lookup("storage-class")
+	if storageClassFlag != nil && storageClassFlag.Changed {
+		storageClass := cmdutil.GetFlagString(cmd, "storage-class")
+		c.StorageClass = &storageClass
+	}
+
+	capacityString := cmdutil.GetFlagString(cmd, "capacity")
+	if len(capacityString) != 0 {
+		quantity, err := resource.ParseQuantity(capacityString)
+		if err != nil {
+			return err
+		}
+
+		c.Capacity = &quantity
+	}
+
+	// Complete other options for Run.
+	c.Mapper, _ = f.Object()
+
+	c.DryRun = cmdutil.GetDryRunFlag(cmd)
```

Comment:
Better to use `cmd.Flags().StringVar(xxx)` to get these values, due to the same scope of `CreatePVCOptions`.

Reply:
some of the flags are added through some helper functions like this: `cmdutil.AddDryRunFlag(cmd)`
I agree that if we can change this to use `StringVar` would be better, maybe change the helper functions?